app/main.py

```diff
@@ -1 +1,19 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import NotWearingMaskError, VaccineError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_to_buy = 0
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
```

Comment:
Returning immediately upon catching a `VaccineError` means that the function will stop processing the rest of the friends as soon as one friend is found not vaccinated. Consider accumulating the result or using a flag to indicate if any friend is not vaccinated, and then return the message after processing all friends.